### ISSUE TITLE
tsql: CREATE TABLE for temporal table cannot be parsed

### ISSUE THREAD

**Issue by fmms** [sqlfluff/sqlfluff#3542]
### Search before asking

- [X] I searched the [issues](https://github.com/sqlfluff/sqlfluff/issues) and found no similar issues.

### What Happened

Since SQL Server 2016, there is a feature called temporal tables, see https://docs.microsoft.com/en-us/sql/relational-databases/tables/temporal-tables?view=sql-server-ver16 .

The example from the documentation cannot be parsed:

```
CREATE TABLE dbo.Employee
(
  [EmployeeID] int NOT NULL PRIMARY KEY CLUSTERED
  , [Name] nvarchar(100) NOT NULL
  , [Position] varchar(100) NOT NULL
  , [Department] varchar(100) NOT NULL
  , [Address] nvarchar(1024) NOT NULL
  , [AnnualSalary] decimal (10,2) NOT NULL
  , [ValidFrom] datetime2 GENERATED ALWAYS AS ROW START
  , [ValidTo] datetime2 GENERATED ALWAYS AS ROW END
  , PERIOD FOR SYSTEM_TIME (ValidFrom, ValidTo)
 )
WITH (SYSTEM_VERSIONING = ON (HISTORY_TABLE = dbo.EmployeeHistory));
```

### Expected Behaviour

Table can be parsed and formatted by sqlfluff.

### Observed Behaviour

One gets 
```
==== parsing violations ====
L:   1 | P:   1 |  PRS | Line 1, Position 1: Found unparsable section: 'CREATE TABLE
                       | dbo.Employee\n(\n [EmployeeI...'
WARNING: Parsing errors found and dialect is set to 'tsql'. Have you configured your dialect correctly?
```

### How to reproduce

```
PS > cat .\sqlfluff-4.config
[sqlfluff]
dialect = tsql
PS > sqlfluff parse .\sqlfluff-4.sql --config .\sqlfluff-4.config -vvv
```

### Dialect

tsql

### Version

```
==== sqlfluff ====
sqlfluff:                1.0.0 python:                 3.10.4
implementation:        cpython verbosity:                   3
dialect:                  tsql templater:               jinja
rules:                                all
```

### Configuration

does not matter

### Are you willing to work on and submit a PR to address the issue?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/sqlfluff/sqlfluff/blob/main/CODE_OF_CONDUCT.md)

**Comment by fmms** [comment]
I believe this issue might be a small subset of the generic #1954

**Comment by keen85** [comment]
Hi @WittierDinosaur & @fmms,
is this really fixed?
We found that this [example from Microsoft Doc](https://learn.microsoft.com/en-us/sql/relational-databases/tables/creating-a-system-versioned-temporal-table?view=sql-server-ver16#creating-a-temporal-table-with-a-default-history-table) still leads to a parsing error in SQL 2.0.2:
```sql
CREATE TABLE Department
(
    DeptID INT NOT NULL PRIMARY KEY CLUSTERED
  , DeptName VARCHAR(50) NOT NULL
  , ManagerID INT NULL
  , ParentDeptID INT NULL
  , ValidFrom DATETIME2 GENERATED ALWAYS AS ROW START NOT NULL
  , ValidTo DATETIME2 GENERATED ALWAYS AS ROW END NOT NULL
  , PERIOD FOR SYSTEM_TIME (ValidFrom, ValidTo)
)
WITH (SYSTEM_VERSIONING = ON (HISTORY_TABLE = dbo.DepartmentHistory));
```
